### PR TITLE
Add UI components

### PR DESCRIPTION
**Pull request by YehorPytomets**
### Changes introduced by this PR:
* Add `UI Components`. They allow writing `HTML` code in `JavaScript` and then reuse them instead of hardcoding.
* Update `gradle` configuration to make app module-based.
* ~~Use `markup builder` to assemble `HTML` parts of code in each component.~~ Reject idea of using builder in order to maintain comprehensiveness of `HTML` code.
* Add `JSDocs`.
* Make `FormFooter` and `FormHeader` to be integral part of `Form` component.
* Add `npm` to project.
* Get rid of secondary rendering when setting component properties.
* Add tests for components.

UPD 24 Mar 2020 @YehorPytomets : Comments of @dkalinin's review of 23 Mar 2020 were considered. PR is ready to be rechecked.
UPD 30 Mar 2020 @YehorPytomets : Merge `markup-bootstrap` into `ui-components` branch.
UPD 31 Mar 2020 @YehorPytomets : Refactor components' code. Add tests.
UPD 1 Apr 2020 @YehorPytomets : PR is ready to review.

### CODE REVIEW

**Comment by dkalinin** [comment]
Please add more descriptive title and description. Use [Markdown](https://guides.github.com/features/mastering-markdown/#examples) and see example of description [in Spine project](https://github.com/SpineEventEngine/core-java/pull/1249).

File: webclient/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import Page from "./components/page.js";
+
+const body = document.getElementsByTagName("body")[0];
+new Page(body, {
```
Comment (dkalinin) [inline webclient/index.js]: Let's have Application class on a top level. Then it creates a page.

File: webclient/validator/validator.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export default class Validator {
+
+    validator
```
Comment (dkalinin) [inline webclient/validator/validator.js]: This is not used.

File: webclient/components/form.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        });
+        this.formInputComponents.push(userNameFormInput);
+
+        const passwordFormInput = new FormInput(this.rootElement, {
+            labelText: "Password",
+            inputId: "password-input",
+            inputName: "password",
+            inputType: "password",
+            inputPlaceholder: "Password",
+            errorMessage: "Password can't be empty and should contain letters and numbers"
```
Comment (dkalinin) [inline webclient/components/form.js]: EmptyValidator doesn't check letters and numbers so let's tell user only about empty input.

**Comment by YehorPytomets** [comment]
@dkalinin, `markup-bootstrap` was merged into `ui-components`. PR is ready to be reviewed again.

**Comment by YehorPytomets** [comment]
PR is ready to review.

**Pull request closed without merging**